### PR TITLE
Upgrade to Spring Boot 2.6.0

### PR DESCRIPTION
**Pull request by ptuomola**
## Description

Describe the changes made and why they were made.

Ignore if these details are present on the associated [Apache Fineract JIRA ticket](https://github.com/apache/fineract/pull/1284).

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Write the commit message as per https://github.com/apache/fineract/#pull-requests

- [x] Acknowledge that we will not review PRs that are not passing the build _("green")_ - it is your responsibility to get a proposed PR to pass the build, not primarily the project's maintainers.

- [x] Create/update unit or integration tests for verifying the changes made.

- [x] Follow coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions.

- [x] Add required Swagger annotation and update API documentation at fineract-provider/src/main/resources/static/api-docs/apiLive.htm with details of any API changes

- [x] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the developer mailing list for guidance, if required.)

FYI our guidelines for code reviews are at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide.

### CODE REVIEW

**Comment by ptuomola** [comment]
@vidakovic This upgrades the Spring Boot to 2.6.0 - an upgrade that was unblocked by the config file rework. If you can approve and merge this, as next step I'll then rework the OAuth and twofactor authentication, and write some test cases for both of them.

File: fineract-provider/src/main/java/org/apache/fineract/infrastructure/security/filter/TenantAwareBasicAuthenticationFilter.java [hunk 1]
```diff
@@ -59,33 +58,34 @@
...
 
     private static boolean firstRequestProcessed = false;
     private static final Logger LOG = LoggerFactory.getLogger(TenantAwareBasicAuthenticationFilter.class);
 
-    private final BasicAuthTenantDetailsService basicAuthTenantDetailsService;
-    private final ToApiJsonSerializer<PlatformRequestLog> toApiJsonSerializer;
-    private final ConfigurationDomainService configurationDomainService;
-    private final CacheWritePlatformService cacheWritePlatformService;
-    private final NotificationReadPlatformService notificationReadPlatformService;
+    @Autowired
```
Comment (vidakovic) [inline fineract-provider/src/main/java/org/apache/fineract/infrastructure/security/filter/TenantAwareBasicAuthenticationFilter.java:68]: A compromise here could be to use Lombok with "RequiredArgsConstructor"... then we can keep the same dependency injection strategy without being annoying (aka having to write all the constructor boilerplate code). Personally the "@Autowired" are OK for me... not sure if others have an opinion here.

If we introduce Lombok (I really would like) then I prefer the combination of "@RequiredArgsConstructor" and final class attributes.

My 2 cents.

**Comment by vidakovic** [comment]
Overall looks good to me. Just one caveat: at the moment there is no compatible Spring Cloud release for Spring Boot 2.6.0. Just in case someone is using that combination in customized projects. For the Fineract release it's irrelevant.

@ptuomola any ideas why the Travis stuff is failing? Other than that I think this can be approved.

**Comment by ptuomola** [comment]
> Overall looks good to me. Just one caveat: at the moment there is no compatible Spring Cloud release for Spring Boot 2.6.0. Just in case someone is using that combination in customized projects. For the Fineract release it's irrelevant.
> 
> @ptuomola any ideas why the Travis stuff is failing? Other than that I think this can be approved.

The GitHub Gradle build runs exactly the same code as the Travis one, so I think we're safe here. Overall I think the Travis is much more susceptible to all kinds of disturbances (load, timing etc) than the GitHub one - and the Github is faster. I think we should reduce the Travis one to just run the SonarQube analysis (as that can't be integrated to GitHub).

File: fineract-provider/src/main/java/org/apache/fineract/infrastructure/security/filter/TenantAwareBasicAuthenticationFilter.java (hunk 1 above)
Comment (ptuomola) [reply fineract-provider/src/main/java/org/apache/fineract/infrastructure/security/filter/TenantAwareBasicAuthenticationFilter.java:68]: I think this is a case of personal preferences... personally I like declaring the variables as @Autowired more than the "huge constructor with lots of parameters" approach, as it's simply less code: you just need to declare the variables, not repeat them three times (one declaration and twice in constructor). But you're right that it differs from how most of the classes are currently written...

**Review by vidakovic — APPROVED** [review]
LGTM

**Pull request merged**